### ISSUE TITLE
Validation IoU drops as training proceeds?

### ISSUE THREAD

**Issue by xiankgx** [stashvala/VideoMatch#2]
Dear authors, I'm trying out the training code on the DAVIS 2017 datset. However, I'm seeing the validation IoU drops as I train. The only changes is I'm using the "Full-Resolution" version of the dataset as opposed to the 480p version.

```
python3 main.py -d ../datasets/DAVIS-2017-trainval-Full-Resolution/DAVIS/ -y 2017 -c 1 -m train -t trainval -s models/vm_vgg.pth -p 416 790 -l 1e-5 -x -uf 2
  DEBUG [       davis.py:061]:	Number of specified sequences in davis dataset is 90
   INFO [     encoder.py:027]:	Using vgg as encoder, with upsample factor 2.0
  DEBUG [       davis.py:133]:	Number of all pairs 224301
  DEBUG [        main.py:171]:	Train set size: 223741, Validation set size: 560
  DEBUG [  preprocess.py:049]:	Using image augmentation with the following transformations: ['hflip', 'random_crop', 'random_scale']
  DEBUG [        main.py:197]:	Number of trainable parameters in VideoMatch: 10585152
  DEBUG [        main.py:211]:	Using foreground threshold 0.5
  DEBUG [        main.py:212]:	Running untrained VideoMatch on validation set...
  DEBUG [        main.py:225]:	Untrained Videomatch IOU on validation set: 0.643
  DEBUG [        main.py:236]:	Using loss function dice
  DEBUG [        main.py:237]:	Training started...
  DEBUG [        main.py:242]:	Epoch: 	[1/1]
[W TensorIterator.cpp:924] Warning: Mixed memory format inputs detected while calling the operator. The operator will output channels_last tensor even if some of the inputs are not in channels_last format. (function operator())
  DEBUG [        main.py:274]:	Iter [   50/223741]:	avg loss = 0.7616,	avg val IOU = 0.667
  DEBUG [        main.py:274]:	Iter [  100/223741]:	avg loss = 0.7402,	avg val IOU = 0.689
  DEBUG [        main.py:274]:	Iter [  150/223741]:	avg loss = 0.7419,	avg val IOU = 0.691
  DEBUG [        main.py:274]:	Iter [  200/223741]:	avg loss = 0.7653,	avg val IOU = 0.703
  DEBUG [        main.py:274]:	Iter [  250/223741]:	avg loss = 0.7854,	avg val IOU = 0.703
  DEBUG [        main.py:274]:	Iter [  300/223741]:	avg loss = 0.7772,	avg val IOU = 0.702
  DEBUG [        main.py:274]:	Iter [  350/223741]:	avg loss = 0.7201,	avg val IOU = 0.710
  DEBUG [        main.py:274]:	Iter [  400/223741]:	avg loss = 0.7278,	avg val IOU = 0.710
  DEBUG [        main.py:274]:	Iter [  450/223741]:	avg loss = 0.7509,	avg val IOU = 0.705
  DEBUG [        main.py:274]:	Iter [  500/223741]:	avg loss = 0.7368,	avg val IOU = 0.701
  DEBUG [        main.py:274]:	Iter [  550/223741]:	avg loss = 0.7425,	avg val IOU = 0.675
  DEBUG [        main.py:274]:	Iter [  600/223741]:	avg loss = 0.7555,	avg val IOU = 0.638
  DEBUG [        main.py:274]:	Iter [  650/223741]:	avg loss = 0.7715,	avg val IOU = 0.547
  DEBUG [        main.py:274]:	Iter [  700/223741]:	avg loss = 0.7599,	avg val IOU = 0.478
  DEBUG [        main.py:274]:	Iter [  750/223741]:	avg loss = 0.7683,	avg val IOU = 0.435
  DEBUG [        main.py:274]:	Iter [  800/223741]:	avg loss = 0.7420,	avg val IOU = 0.440
  DEBUG [        main.py:274]:	Iter [  850/223741]:	avg loss = 0.7805,	avg val IOU = 0.445
  DEBUG [        main.py:274]:	Iter [  900/223741]:	avg loss = 0.7752,	avg val IOU = 0.390
  DEBUG [        main.py:274]:	Iter [  950/223741]:	avg loss = 0.7028,	avg val IOU = 0.367
  DEBUG [        main.py:274]:	Iter [ 1000/223741]:	avg loss = 0.7668,	avg val IOU = 0.345
  DEBUG [        main.py:274]:	Iter [ 1050/223741]:	avg loss = 0.7634,	avg val IOU = 0.311
  DEBUG [        main.py:274]:	Iter [ 1100/223741]:	avg loss = 0.7538,	avg val IOU = 0.347
  DEBUG [        main.py:274]:	Iter [ 1150/223741]:	avg loss = 0.7521,	avg val IOU = 0.312
  DEBUG [        main.py:274]:	Iter [ 1200/223741]:	avg loss = 0.7957,	avg val IOU = 0.252
  DEBUG [        main.py:274]:	Iter [ 1250/223741]:	avg loss = 0.7770,	avg val IOU = 0.217
  DEBUG [        main.py:274]:	Iter [ 1300/223741]:	avg loss = 0.7699,	avg val IOU = 0.204
  DEBUG [        main.py:274]:	Iter [ 1350/223741]:	avg loss = 0.7779,	avg val IOU = 0.184
  DEBUG [        main.py:274]:	Iter [ 1400/223741]:	avg loss = 0.7962,	avg val IOU = 0.181
  DEBUG [        main.py:274]:	Iter [ 1450/223741]:	avg loss = 0.7665,	avg val IOU = 0.176
  DEBUG [        main.py:274]:	Iter [ 1500/223741]:	avg loss = 0.7732,	avg val IOU = 0.178
  DEBUG [        main.py:274]:	Iter [ 1550/223741]:	avg loss = 0.7985,	avg val IOU = 0.176
  DEBUG [        main.py:274]:	Iter [ 1600/223741]:	avg loss = 0.8092,	avg val IOU = 0.175
  DEBUG [        main.py:274]:	Iter [ 1650/223741]:	avg loss = 0.7817,	avg val IOU = 0.175
  DEBUG [        main.py:274]:	Iter [ 1700/223741]:	avg loss = 0.7861,	avg val IOU = 0.175
  DEBUG [        main.py:274]:	Iter [ 1750/223741]:	avg loss = 0.7919,	avg val IOU = 0.169
  DEBUG [        main.py:274]:	Iter [ 1800/223741]:	avg loss = 0.8039,	avg val IOU = 0.170
  DEBUG [        main.py:274]:	Iter [ 1850/223741]:	avg loss = 0.7818,	avg val IOU = 0.170
  DEBUG [        main.py:274]:	Iter [ 1900/223741]:	avg loss = 0.8001,	avg val IOU = 0.171
  DEBUG [        main.py:274]:	Iter [ 1950/223741]:	avg loss = 0.8212,	avg val IOU = 0.180
  DEBUG [        main.py:274]:	Iter [ 2000/223741]:	avg loss = 0.7570,	avg val IOU = 0.180
  DEBUG [        main.py:274]:	Iter [ 2050/223741]:	avg loss = 0.7544,	avg val IOU = 0.179

```

**Comment by stashvala** [comment]
@xiankgx Since you are using DAVIS 2017, my first guess would be that multiple instances/classes are the problem. You could try training the model using only one class from the masks, ergo single object segmentation. You can extend this idea to evaluation, where you run the model for each class in the image separately and thus constructing multiple object segmentation.

**Comment by xiankgx** [comment]
Ok, I totally missed the single vs multiple instances segmentation here. Thank you author, your quick response and support is much appreciated.

**Comment by 863689877** [comment]
Dear author, I have faced the similar problem when I train the net on davis 2016. 
I used `python main.py -c 0 -m train -t train -s models/vm_vgg.pth -p 416 790 -l 1e-5 -x` to start training, and got the log as follow.
```
DEBUG [       davis.py:062]:	Number of specified sequences in davis dataset is 30
   INFO [     encoder.py:027]:	Using vgg as encoder, with upsample factor 1.0
  DEBUG [       davis.py:135]:	Number of all pairs 74602
  DEBUG [        main.py:171]:	Train set size: 74416, Validation set size: 186
  DEBUG [  preprocess.py:049]:	Using image augmentation with the following transformations: ['hflip', 'random_crop', 'random_scale']
  DEBUG [        main.py:197]:	Number of trainable parameters in VideoMatch: 10585152
  DEBUG [        main.py:211]:	Using foreground threshold 0.5
  DEBUG [        main.py:212]:	Running untrained VideoMatch on validation set...
  DEBUG [        main.py:225]:	Untrained Videomatch IOU on validation set: 0.638
  DEBUG [        main.py:236]:	Using loss function dice
  DEBUG [        main.py:237]:	Training started...
  DEBUG [        main.py:242]:	Epoch: 	[1/1]
[W TensorIterator.cpp:924] Warning: Mixed memory format inputs detected while calling the operator. The operator will output channels_last tensor even if some of the inputs are not in channels_last format. (function operator())
  DEBUG [        main.py:274]:	Iter [   50/74416]:	avg loss = 0.8048,	avg val IOU = 0.720
  DEBUG [        main.py:274]:	Iter [  100/74416]:	avg loss = 0.7860,	avg val IOU = 0.752
  DEBUG [        main.py:274]:	Iter [  150/74416]:	avg loss = 0.7890,	avg val IOU = 0.757
  DEBUG [        main.py:274]:	Iter [  200/74416]:	avg loss = 0.7743,	avg val IOU = 0.770
  DEBUG [        main.py:274]:	Iter [  250/74416]:	avg loss = 0.7817,	avg val IOU = 0.769
  DEBUG [        main.py:274]:	Iter [  300/74416]:	avg loss = 0.7906,	avg val IOU = 0.769
  DEBUG [        main.py:274]:	Iter [  350/74416]:	avg loss = 0.7737,	avg val IOU = 0.733
  DEBUG [        main.py:274]:	Iter [  400/74416]:	avg loss = 0.7693,	avg val IOU = 0.476
  DEBUG [        main.py:274]:	Iter [  450/74416]:	avg loss = 0.7676,	avg val IOU = 0.288
  DEBUG [        main.py:274]:	Iter [  500/74416]:	avg loss = 0.8078,	avg val IOU = 0.216
  DEBUG [        main.py:274]:	Iter [  550/74416]:	avg loss = 0.7770,	avg val IOU = 0.181
  DEBUG [        main.py:274]:	Iter [  600/74416]:	avg loss = 0.8047,	avg val IOU = 0.152
  DEBUG [        main.py:274]:	Iter [  650/74416]:	avg loss = 0.7901,	avg val IOU = 0.191
  DEBUG [        main.py:274]:	Iter [  700/74416]:	avg loss = 0.8225,	avg val IOU = 0.174
  DEBUG [        main.py:274]:	Iter [  750/74416]:	avg loss = 0.8217,	avg val IOU = 0.169
  DEBUG [        main.py:274]:	Iter [  800/74416]:	avg loss = 0.8020,	avg val IOU = 0.187
  DEBUG [        main.py:274]:	Iter [  850/74416]:	avg loss = 0.7928,	avg val IOU = 0.191
  DEBUG [        main.py:274]:	Iter [  900/74416]:	avg loss = 0.8316,	avg val IOU = 0.184
  DEBUG [        main.py:274]:	Iter [  950/74416]:	avg loss = 0.8386,	avg val IOU = 0.198
  DEBUG [        main.py:274]:	Iter [ 1000/74416]:	avg loss = 0.8094,	avg val IOU = 0.183
  DEBUG [        main.py:274]:	Iter [ 1050/74416]:	avg loss = 0.7938,	avg val IOU = 0.192
  DEBUG [        main.py:274]:	Iter [ 1100/74416]:	avg loss = 0.7821,	avg val IOU = 0.182
  DEBUG [        main.py:274]:	Iter [ 1150/74416]:	avg loss = 0.8171,	avg val IOU = 0.178
  DEBUG [        main.py:274]:	Iter [ 1200/74416]:	avg loss = 0.7996,	avg val IOU = 0.137
  DEBUG [        main.py:274]:	Iter [ 1250/74416]:	avg loss = 0.8253,	avg val IOU = 0.130
  DEBUG [        main.py:274]:	Iter [ 1300/74416]:	avg loss = 0.8318,	avg val IOU = 0.127
  DEBUG [        main.py:274]:	Iter [ 1350/74416]:	avg loss = 0.8340,	avg val IOU = 0.153
  DEBUG [        main.py:274]:	Iter [ 1400/74416]:	avg loss = 0.8016,	avg val IOU = 0.171
  DEBUG [        main.py:274]:	Iter [ 1450/74416]:	avg loss = 0.8026,	avg val IOU = 0.158
  DEBUG [        main.py:274]:	Iter [ 1500/74416]:	avg loss = 0.8261,	avg val IOU = 0.156
  DEBUG [        main.py:274]:	Iter [ 1550/74416]:	avg loss = 0.8373,	avg val IOU = 0.159
  DEBUG [        main.py:274]:	Iter [ 1600/74416]:	avg loss = 0.8490,	avg val IOU = 0.156
  DEBUG [        main.py:274]:	Iter [ 1650/74416]:	avg loss = 0.8260,	avg val IOU = 0.158
  DEBUG [        main.py:274]:	Iter [ 1700/74416]:	avg loss = 0.8072,	avg val IOU = 0.151
  DEBUG [        main.py:274]:	Iter [ 1750/74416]:	avg loss = 0.8316,	avg val IOU = 0.138
  DEBUG [        main.py:274]:	Iter [ 1800/74416]:	avg loss = 0.8281,	avg val IOU = 0.154
  DEBUG [        main.py:274]:	Iter [ 1850/74416]:	avg loss = 0.8112,	avg val IOU = 0.144
  DEBUG [        main.py:274]:	Iter [ 1900/74416]:	avg loss = 0.8384,	avg val IOU = 0.143
  DEBUG [        main.py:274]:	Iter [ 1950/74416]:	avg loss = 0.8355,	avg val IOU = 0.141
  DEBUG [        main.py:274]:	Iter [ 2000/74416]:	avg loss = 0.8488,	avg val IOU = 0.139
  DEBUG [        main.py:274]:	Iter [ 2050/74416]:	avg loss = 0.8240,	avg val IOU = 0.136
```
